server/src/modules/auth/decorators/has-roles.decorator.ts

```diff
@@ -0,0 +1,4 @@
+import { SetMetadata } from '@nestjs/common';
+import { Roles } from 'src/common/enums';
+
+export const HasRoles = (role: Roles) => SetMetadata('role', role);
```

Comment:
I suggest using rest operator for roles parameter to be able to use many roles restrictions for endpoints.

And after that, we need to change RolesGuard logic to work with array of roles instead of single role
```suggestion
export const HasRoles = (...roles: UserRoleEnum[]) => SetMetadata('roles', roles);
```